src/Error/ExceptionRenderer.php

```diff
@@ -125,9 +125,18 @@ protected function _unwrap($exception)
      */
     protected function _getController()
     {
-        $request = $this->request ?: Router::getRequest(true);
+        $request = $this->request;
+        $routerRequest = Router::getRequest(true);
+        // Fallback to the request in the router or make a new one from
+        // $_SERVER
         if ($request === null) {
-            $request = ServerRequestFactory::fromGlobals();
+            $request = $routerRequest ?: ServerRequestFactory::fromGlobals();
+        }
+
+        // If the current request doesn't have routing data, but we
+        // found a request in the router context copy the params over
+        if ($request->getParam('controller') === false && $routerRequest !== null) {
+            $request = $request->withAttribute('params', $routerRequest->getAttribute('params'));
```

Comment:
why not whole request ?
```
if ($request->getParam('controller') === false && $routerRequest !== null) {
            $request =$routerRequest;
```

Reply:
I was concerned with overwriting headers or transformed data.